handlers/update.go

```diff
@@ -28,81 +28,123 @@ func MakeUpdateHandler(functionNamespace string, clientset *kubernetes.Clientset
 			return
 		}
 
-		getOpts := metav1.GetOptions{}
-
-		deployment, findDeployErr := clientset.ExtensionsV1beta1().
-			Deployments(functionNamespace).
-			Get(request.Service, getOpts)
+		annotations := buildAnnotations(request)
+		if err, status := updateDeploymentSpec(functionNamespace, clientset, request, annotations); err != nil {
+			w.WriteHeader(status)
+			w.Write([]byte(err.Error()))
+		}
 
-		if findDeployErr != nil {
-			w.WriteHeader(http.StatusNotFound)
-			w.Write([]byte(findDeployErr.Error()))
-			return
+		if err, status := updateService(functionNamespace, clientset, request, annotations); err != nil {
+			w.WriteHeader(status)
+			w.Write([]byte(err.Error()))
 		}
+	}
+}
 
-		if len(deployment.Spec.Template.Spec.Containers) > 0 {
-			deployment.Spec.Template.Spec.Containers[0].Image = request.Image
+func updateDeploymentSpec(
+	functionNamespace string,
```

Comment:
@LucasRoesler @alexellis i've *not* added a test to this method as its not unit testable (as a side not it's not new code just extracted into a separate func from MakeUpdateHandler, to hopefully make MakeUpdateHandler easier to follow)

Reply:
Technically it is possible to test it by using the `fake` package https://godoc.org/k8s.io/client-go/kubernetes/fake#Clientset but it probably makes sense to consider the design of the function to avoid that if we can and that can be done in the future so that we avoid changing too many things in this PR